numba/core/compiler.py

```diff
@@ -144,6 +144,20 @@ class Flags(TargetConfig):
         doc="backend"
     )
 
+    dbg_extend_lifetimes = Option(
+        type=bool,
+        default=False,
+        doc=("Extend variable lifetime for debugging. "
+             "This automatically turns on with debug=True."),
```

Comment:
Whilst not introduced by this PR, it seems confusing that user flag `debug` maps to `debuginfo` on the target. But `debuginfo` kind of implies DI for DWARF emission.

Reply:
Yes, this goes back to the discussion to better define what `debug` flag does.